portal-web/docroot/html/taglib/ui/page_iterator/lexicon/start.jsp

```diff
@@ -338,7 +338,7 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 
 <c:if test="<%= pages > initialPages %>">
 	<aui:script require="frontend-js-web/liferay/DynamicInlineScroll.es">
-		new frontendJsWebLiferayDynamicInlineScrollEs.default(
+		var dynamicInlineScrollHandler = new frontendJsWebLiferayDynamicInlineScrollEs.default(
```

Comment:
Since we're doing a `new` and want to handle lifecycle, this sounds like a good opportunity to use `liferay-frontend:component` which does exactly that.

Reply:
Unfortunately, this is in portal-web, so we don't have access to any of the classes in com.liferay.frontend.taglib, which lives outside of the Liferay .war

Reply:
When testing this change, I noticed a console error coming from the earlier changes to `detached`.

Reading through the function that was throwing it, I found that `addEventListener` returns nothing, so I needed to fix that. I also noticed some weirdness in the logic in the same `detached` function (it was creating a new list item and then detaching listeners from it), so I rewrote it to match what I *think* it was supposed to be doing.